src/util.cpp

```diff
@@ -371,6 +371,7 @@ void ParseParameters(int argc, const char* const argv[])
         if (str.length() > 1 && str[1] == '-')
             str = str.substr(1);
         InterpretNegativeSetting(str, strValue);
+        checkArgFunc(str.substr(1));
```

Comment:
I don't think that check is necessary here, because a few lines above is:

```
if (str[0] != '-')
    break;
```

So, at this point, we know the string is at least length 1.